File: mobile/android/app/src/main/java/com/findly/android/ui/designsystem/components/FindlySectionHeader.kt
package com.findly.android.ui.designsystem.components

import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.findly.android.ui.designsystem.FindlyTheme

/**
 * A stateless section-title label (grouping a list — e.g. "Devices"/"Members" on the A2 settings
 * screen). Added in A2 so a screen never needs to reach for a bare Material3 `Text` composable
 * directly (specs/003-android-client.md §4.3: screens compose only `ui/designsystem` components).
 * Reads only [FindlyTheme] tokens. Geometry from design 2a
 * (design/findly-design-system/2a-ember-dusk/HANDOFF.md, `FindlySectionHeader`): `labelSmall`
 * (uppercased — `TextStyle` has no text-transform, so [title] is uppercased here), muted color,
 * 4dp horizontal padding, 10dp below.
 */
@Composable
fun FindlySectionHeader(
    title: String,
    modifier: Modifier = Modifier,
) {
    Text(
        text = title.uppercase(),
        color = FindlyTheme.colors.subtleText,
        style = FindlyTheme.typography.labelSmall,
        modifier = modifier
            .fillMaxWidth()
            .padding(start = FindlyTheme.spacing.xs, end = FindlyTheme.spacing.xs, bottom = 10.dp),
    )
}

File: mobile/ios/FindlyKit/Sources/FindlyKit/DesignSystem/Components/FindlySectionHeader.swift
import SwiftUI

/// specs/004-ios-client.md §2.3 — a section label (e.g. "MEMBERS", "DEVICES" on Family & devices).
/// New in I27: design 2a "Ember/Dusk" (design/findly-design-system/2a-ember-dusk/HANDOFF.md) is
/// the first handoff to specify this as its own named component rather than ad hoc screen text.
///
/// `labelSmall`, uppercase, muted, 4pt horizontal padding (== `SpacingTokens.xs`). The handoff's
/// "10pt below the previous block" is spacing between this header and whatever precedes it in a
/// screen's layout — a caller concern (`Screens/`, I28), not something this stateless component
/// can apply to itself.
public struct FindlySectionHeader: View {
    @Environment(\.theme) private var theme
    private let title: String

    public init(_ title: String) {
        self.title = title
    }

    public var body: some View {
        Text(title)
            .font(theme.typography.labelSmall.font)
            .tracking(theme.typography.labelSmall.tracking)
            .textCase(.uppercase)
            .foregroundColor(theme.onSurfaceMuted)
            .padding(.horizontal, theme.spacing.xs)
    }
}

#Preview("FindlySectionHeader — light") {
    VStack(alignment: .leading, spacing: 8) {
        FindlySectionHeader("Members")
        FindlySectionHeader("Devices")
    }
    .padding()
    .background(Theme.light.colors.surface)
    .environment(\.theme, .light)
}

#Preview("FindlySectionHeader — dark") {
    VStack(alignment: .leading, spacing: 8) {
        FindlySectionHeader("Members")
        FindlySectionHeader("Devices")
    }
    .padding()
    .background(Theme.dark.colors.surface)
    .environment(\.theme, .dark)
    .preferredColorScheme(.dark)
}
